src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt

```diff
@@ -145,19 +148,22 @@ private fun Project.configureCheckTasks(
     }
 
     task<Sync>("apiDump") {
-        isEnabled = apiCheckEnabled(extension) && apiBuild.map { it.enabled }.getOrElse(true)
+        // Do not execute task for empty umbrella modules
+        setOnlyIf { apiBuild.get().isOnlyIf() }
+        isEnabled = apiCheckEnabled(extension)
         group = "other"
         description = "Syncs API from build dir to $API_DIR dir for $projectName"
         from(apiBuildDir)
         into(apiCheckDir)
         dependsOn(apiBuild)
-        doFirst {
-            apiCheckDir.mkdirs()
-        }
+        doFirst { apiCheckDir.mkdirs() }
     }
     project.tasks.getByName("check").dependsOn(apiCheck)
 }
 
+// the task handed to isSatisfiedBy is never used, so it does not matter what we hand over.
```

Comment:
I think passing `this` is actually the correct task. The gradle api is old java code, so this is intended to be some class in some other file, hence they pass the `Task` into the callback. with kotlin you almost never need it, because if used with a lambda, you capture it anyways. TL/DR: I'd just remove that comment.